Sources/ArcGISToolkit/FloorFilter/FloorFilterViewModel.swift

```diff
@@ -0,0 +1,118 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import UIKit
+import ArcGIS
+
+// View Model class that contains the Data Model of the Floor Filter
+// Also contains the business logic to filter and change the map extent based on selected site/level/facility
+internal class FloorFilterViewModel {
+    
+    // The MapView, Map and Floor Manager are set in the FloorFilterViewController when the map is loaded
+    public var mapView: AGSMapView?
+    public var map: AGSMap?
+    public var floorManager: AGSFloorManager?
+    
+    public var sites: [AGSFloorSite] {
+        return floorManager?.sites ?? []
+    }
+    
+    // Facilities in the site that is selected
+    // If no site is selected then the list is empty
+    // If the sites data does not exist in the map, then use all the facilities in the map
+    public var facilities: [AGSFloorFacility] {
+        guard let floorManager = floorManager else { return [] }
+        return sites.isEmpty ? floorManager.facilities : floorManager.facilities.filter { $0.site == selectedSite }
+    }
+    
+    // Levels that are visible in the expanded Floor Filter levels table view
+    // Reverse the order of the levels to make it in ascending order
+    public var visibleLevelsInExpandedList: [AGSFloorLevel] {
+        guard let floorManager = floorManager else { return [] }
+        return facilities.isEmpty ? floorManager.levels : floorManager.levels.filter { $0.facility == selectedFacility }.reversed()
+    }
+    
+    // All the levels in the map
+    public var allLevels: [AGSFloorLevel] {
+        return floorManager?.levels ?? []
+    }
+    
+    // The site, facility, and level that are selected by the user
+    public var selectedSite: AGSFloorSite?
+    public var selectedFacility: AGSFloorFacility?
+    public var selectedLevel: AGSFloorLevel?
+    
+    // The default vertical order is 0 according to Runtime 100.12 update for AGSFloorManager
+    public let defaultVerticalOrder = 0
+
+    public func reset() {
+        floorManager = nil
+        selectedSite = nil
+        selectedFacility = nil
+        selectedLevel = nil
+    }
+    
+    public func getSelectedSite() -> AGSFloorSite? {
+        return sites.first { $0 == selectedSite }
+    }
+
+    public func getSelectedFacility() -> AGSFloorFacility? {
+        return facilities.first { $0 == selectedFacility }
+    }
+    
+    public func getSelectedVisibleLevel() -> AGSFloorLevel? {
+        return visibleLevelsInExpandedList.first { $0 == selectedLevel }
+    }
+    
+    // Sets the visibility of all the levels on the map based on the vertical order of the current selected level
+    public func filterMapToSelectedLevel() {
+        guard let selectedLevel = getSelectedVisibleLevel() ?? selectedLevel else { return }
+        allLevels.forEach {
+            $0.isVisible = $0.verticalOrder == selectedLevel.verticalOrder
+        }
```

Comment:
Can a level other than the `selectedLevel` have a `verticalOrder` value the same as `selectedLevel.verticalOrder`?  Meaning, can this be written as this?
```suggestion
        allLevels.forEach { $0.isVisible = $0 == selectedLevel }
```

Reply:
Unfortunately, doing that does not filter levels in other facilities. Those levels cannot be directly compared to the Level object, but instead by the vertical order. The levels are all unique, the only common element in the vertical order.

Reply:
Unfortunately, doing that does not filter levels in other facilities. Those levels cannot be directly compared to the Level object, but instead by the vertical order. The levels are all unique, the only common element in the vertical order.